enn.py

```diff
@@ -33,41 +36,112 @@ def __init__(self, *args, **kwargs):
+            losses_per_layers_student_mean_valid = {module_name: [(sum(losses_per_layers_student_valid[module_name][i]) / len(losses_per_layers_student_valid[module_name][i])).item() for i in range(0, self.args.number_of_rsacm)] for module_name in self.model.replaced_modules}
 
-                loss_distill = 0
-                for index, module_name in enumerate(self.model.replaced_modules):
-                    get_module_by_name(self.model, module_name)(self.teacher_model.inputs[index][0])
-                    for rsacm_index, rsacm_block_output in enumerate(get_module_by_name(self.model, module_name).x_list):
-                        loss_distill += self.criterion_distilation(rsacm_block_output, self.teacher_model.activation[index])
-                        losses_per_layers_student[module_name][rsacm_index].append(self.criterion_distilation(rsacm_block_output, self.teacher_model.activation[index]))
-        # print(losses_per_layers_student)
-        losses_per_layers_student_mean = {module_name: [(sum(losses_per_layers_student[module_name][i]) / len(losses_per_layers_student[module_name][i])).item() for i in range(0, self.args.number_of_rsacm)] for module_name in self.model.replaced_modules}
-        print(losses_per_layers_student_mean)
-        for module_name in losses_per_layers_student_mean:
-            # Make a random dataset:
-            rsacms = range(1, self.args.number_of_rsacm+1)
-            loss_value = losses_per_layers_student_mean[module_name]
-            y_pos = np.arange(len(rsacms))
-
-            # Create bars
-            plt.bar(y_pos, loss_value)
-
-            # Create names on the x-axis
-            plt.xticks(y_pos, rsacms)
-            plt.title(f'MSELoss in {module_name}')
-            plt.plot()
-            plt.xlabel('Number of RSACM')
-            plt.ylabel('MSELoss value')
-            plt.savefig(f'./graphs/{module_name}.png')
-            plt.clf()
+            losses_per_layers_student_train = {module_name: [[] for i in range(0,self.args.number_of_rsacm)] for module_name in self.model.replaced_modules}
+            for i, (images, labels) in enumerate(tqdm(train_loader)):
+                    images, labels = images.to(self.args.device), labels.to(self.args.device)
+                    self.teacher_model(images)
+                    loss_distill = 0
+                    for index, module_name in enumerate(self.model.replaced_modules):
+                        get_module_by_name(self.model, module_name)(self.teacher_model.inputs[index][0])
+                        for rsacm_index, rsacm_block_output in enumerate(get_module_by_name(self.model, module_name).x_list):
+                            loss_distill += self.criterion_distilation(rsacm_block_output, self.teacher_model.activation[index])
+                            losses_per_layers_student_train[module_name][rsacm_index].append(self.criterion_distilation(rsacm_block_output, self.teacher_model.activation[index]))
```

Comment:
Minor, but duplicated code similar to lines 53-65, perhaps extract it to a function and run it once for train_loader and second time for valid_loader?